app/grandchallenge/reader_studies/tasks.py

```diff
@@ -79,17 +84,17 @@ def add_image_to_display_set(
             interface=interface, image=image
         ).first()
         if civ is None:
-            civ = ComponentInterfaceValue(interface=interface)
-        civ.image = image
-        try:
-            civ.full_clean()
-        except ValidationError as e:
-            upload_session.status = RawImageUploadSession.FAILURE
-            upload_session.error_message = e.message
-            upload_session.save()
-        else:
-            civ.save()
-            display_set.values.add(civ)
+            civ = ComponentInterfaceValue.objects.create(
```

Comment:
Should this be a get_or_create?

Reply:
On this specific line no, but two lines above, we should probably better use a get_or_create, saves some lines of code as well